ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -394,15 +394,22 @@ static const uint32_t private_key_opt_or_null[] = {
 };
 
 /* PKCS#11 specification for any RSA key (+public/private_key_xxx) */
-static const uint32_t rsa_public_key_mandated[] = {
+static const uint32_t rsa_pub_key_gen_mand[] = {
 	PKCS11_CKA_MODULUS_BITS,
 };
 
-static const uint32_t rsa_public_key_opt_or_null[] = {
+static const uint32_t rsa_pub_key_create_mand[] = {
 	PKCS11_CKA_MODULUS, PKCS11_CKA_PUBLIC_EXPONENT,
 };
 
-static const uint32_t rsa_private_key_opt_or_null[] = {
+static const uint32_t rsa_pub_key_gen_opt[] = {
+	PKCS11_CKA_PUBLIC_EXPONENT,
+};
+
+static const uint32_t rsa_pub_key_create_opt[] = {
+};
+
+static const uint32_t rsa_priv_key_opt[] = {
```

Comment:
rename this `rsa_priv_key_create_opt`

Reply:
They are common for both create and generate -- eg. any RSA private key should have them.

If you want to emphasize that then perhaps `rsa_priv_key_common_opt` (or `rsa_priv_key_common_oon`).

I would keep it as is but can modify it if it makes it clearer for reader.

Which way to go?

Reply:
Based on your input I suppose there should be `_opt_or_null` somehow also here if they end up in matching function.

Reply:
Updated them. Please check it out.